File: tutorials/testing/python/tut_sllist_p2.py
from bridges.bridges import *
from bridges.sl_element import *
import sys

#
# In part 2 of this tutorial we demonstrate how we can style the nodes and links
# of the singly linked list  and displays them
#
# Reference: SLelement, Bridges classes, ElementVisualizer, LinkVisualizer, Color
#

def main():

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # set title, description
    bridges.set_title("A Singly Linked List Example")
    bridges.set_description("Demonstrate styling linked list nodes and links");

	# create the list elements
    st0 = SLelement(e="Gretel Chaney")
    st1 = SLelement(e="Lamont Kyler")
    st2 = SLelement(e="Gladys Serino")
    st3 = SLelement(e="Karol Soderman")
    st4 = SLelement(e="Starr McGinn")

    # link the elements
    st0.next = st1
    st1.next = st2
    st2.next = st3
    st3.next = st4

    # we want to see these names in the visualization so we will 
    #set them as the nodes' labels. We will retrieve the nodes' 
    #generic data for the label
    st0.label = st0.value
    st1.label = st1.value
    st2.label = st2.value
    st3.label = st3.value
    st4.label = st4.value


    # now we want to style the nodes and links to illustrate the visual
    # attributes that can be applied to elements and links between links
    # Elements have color, opacity, shape and opacity attributes. Links
    # have color, thickness and label attributes

    # set colors for list elements - see the Color class for supported colors
    st0.color = "red";
    st2.color = "aliceblue";
    
    # color the links - must specify a valid terminating element
    st0.get_link_visualizer(st1).color = "green"
    st3.get_link_visualizer(st4).color = "magenta"
    
    # adjust link thickness
    st2.get_link_visualizer(st3).thickness = 5.0
    
    # set node transparency
    st4.opacity = 0.5
    
    # set link labels
    st0.get_link_visualizer(st1).label = "Gretel Chaney --> Lamont Kyler";
    
    # set shape of node
    st2.shape = "square";
    st4.shape = "triangle";


    # tell Bridges the head of the list
    bridges.set_data_structure(st0)

    # visualize
    bridges.visualize()


if __name__ == "__main__":
    main()

File: tutorials/testing/c++/tut_sllist_p2.cpp
/**
 * Created by Kalpathi Subramanian, 1/30/18
 * krs@uncc.edu
 */

#include "Bridges.h"
#include "SLelement.h"
#include <string>

using namespace bridges;
using namespace std;

//
// In part 2 of this tutorial we demonstrate how we can style the nodes and links
// of the singly linked list  and displays them
//
// Reference: SLelement, Bridges classes, ElementVisualizer, LinkVisualizer, Color
//
int main (int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
	// set title
	bridges.setTitle("A Singly Linked List Example");

	// set  description
	bridges.setDescription("Demonstrate styling linked list nodes and links"); 

	// create the linked list elements with student names
	SLelement<string>  *st0 = new SLelement<string> ("Gretel Chaney");
	SLelement<string>  *st1 = new SLelement<string> ("Lamont Kyler");
	SLelement<string>  *st2 = new SLelement<string> ("Gladys Serino");
	SLelement<string>  *st3 = new SLelement<string> ("Karol Soderman");
	SLelement<string>  *st4 = new SLelement<string> ("Starr McGinn");

	// we want to see these names in the visualization so we will set them as
	// the nodes' labels. We will retrieve the nodes' generic data for the label
	st0->setLabel(st0->getValue());
	st1->setLabel(st1->getValue());
	st2->setLabel(st2->getValue());
	st3->setLabel(st3->getValue());
	st4->setLabel(st4->getValue());

	//  link the elements into a list
	st0->setNext(st1);
	st1->setNext(st2);
	st2->setNext(st3);
	st3->setNext(st4);

	// now we want to style the nodes and links to illustrate the visual
	// attributes that can be applied to elements and links between links
    // Elements have color, opacity, shape and opacity attributes. Links
	// have color, thickness and label attributes

	 // set colors for list elements - see the Color class for supported colors
    st0->setColor("red");
    st2->setColor("aliceblue");

    // color the links - must specify a valid terminating element
    st0->getLinkVisualizer(st1)->setColor("green");
    st3->getLinkVisualizer(st4)->setColor("magenta");

    // adjust link thickness
    st2->getLinkVisualizer(st3)->setThickness(5.0f);

    // set node transparency
    st4->setOpacity (0.5f);

	// set link labels
    st0->getLinkVisualizer(st1)->setLabel("Gretel Chaney --> Lamont Kyler");

	// set shape of node
    st2->setShape(SQUARE);
	st4->setShape (TRIANGLE);

	// tell Bridges the head of the list
	bridges.setDataStructure(st0);

    // visualize the linked list (hit the key 'l' on the visualzation to see all labels
	bridges.visualize();

	//cleanup
	delete st0;
	delete st1;
	delete st2;
	delete st3;
	delete st4;
}
